app/src/components/content/RelatedContent.vue

```diff
@@ -0,0 +1,54 @@
+<script setup lang="ts">
+import IgnorePagePadding from "@/components/IgnorePagePadding.vue";
+import HorizontalScrollableTagViewer from "@/components/tags/HorizontalScrollableTagViewer.vue";
+import { appLanguageIdAsRef } from "@/globalConfig";
+import { db, DocType, type ContentDto, type TagDto, type Uuid } from "luminary-shared";
+import { ref, toRefs, watch } from "vue";
+
+type Props = {
+    tags: TagDto[];
+    currentContentId: Uuid;
+};
+const props = defineProps<Props>();
+
+const { tags } = toRefs(props);
+const content = ref<ContentDto[]>([]);
+
+// Function to fetch content based on tags
+async function fetchContentForTags() {
+    const tagIds = tags.value.map((t: TagDto) => t._id);
+    const contentPromises = tagIds.map((tagId) =>
+        db.contentWhereTag(tagId, { languageId: appLanguageIdAsRef.value }),
+    );
+
+    content.value = (await Promise.all(contentPromises)).flat();
+}
```

Comment:
Instead of running an additional query for each tag, can't we fetch all the content at once?

I'm also wondering why this even happens. It's only used for a check below to check if there is content. But isn't that always the case? I don't think there should be any orphaned tags (i.e. tags without any content) in the system.

Reply:
> I'm also wondering why this even happens. It's only used for a check below to check if there is content. But isn't that always the case? I don't think there should be any orphaned tags (i.e. tags without any content) in the system.

As discussed, this only happens because otherwise we show an empty "related content" if the current post only belongs to one topic with this one post in it.

Reply:
> Instead of running an additional query for each tag, can't we fetch all the content at once?

This still can be fixed - the content is also fetched in `SingleContent`, so now we're doing it twice.